src/composables/transactionList.ts

```diff
@@ -0,0 +1,329 @@
+  return {
+    loaded: [],
+    nextPageUrl: '',
+    pending: {},
+    tipWithdrawnTransactions: [],
+  };
+}
+
+function getAccountTransactionsState(address: Encoded.AccountAddress): ITransactionsState {
+  return transactions.value[address] || generateEmptyTransactionState();
+}
+
+function ensureAccountTransactionStateExists(address: Encoded.AccountAddress) {
+  if (!transactions.value[address]) {
+    transactions.value[address] = generateEmptyTransactionState();
+  }
+}
+
+function getTransactionByHash(address: Encoded.AccountAddress, hash: string) {
+  return transactions.value[address]?.loaded.find((transaction) => transaction.hash === hash);
+}
+
+function updateAccountTransactions(address: Encoded.AccountAddress, transaction: ITransaction) {
+  const transactionIndex = transactions.value[address]?.loaded?.findIndex(
+    (tr) => tr.hash === transaction.hash,
+  );
+
+  if (transactionIndex !== null && transactionIndex >= 0) {
+    transactions.value[address].loaded[transactionIndex] = transaction;
+  } else {
+    ensureAccountTransactionStateExists(address);
+    transactions.value[address].loaded.push(transaction);
+  }
+}
+
+const setTransactionsNextPage = (address: Encoded.AccountAddress, url: string) => {
+  transactions.value[address].nextPageUrl = url;
+};
+
+export function useTransactionList({ store }: IDefaultComposableOptions) {
+  const { nodeNetworkId, getAeSdk } = useAeSdk({ store });
+  const { isLoggedIn } = useAccounts({ store });
+  const {
+    fetchFromMiddlewareCamelCased,
+    getMiddleware,
+  } = useMiddleware({ store });
+
+  const getAccountAllTransactions = (address: Encoded.AccountAddress) => {
+    if (!isLoggedIn) {
+      return [];
+    }
+    const { pending, loaded } = getAccountTransactionsState(address);
+
+    return [...loaded, ...(pending[nodeNetworkId.value!] || [])];
+  };
+
+  function setPendingTransactionSentByHash(address: Encoded.AccountAddress, hash: string) {
+    const index = transactions.value[address].pending[nodeNetworkId.value!]
+      ?.findIndex((transaction) => transaction.hash === hash);
+
+    if (index !== -1) {
+      transactions.value[address]
+        .pending[nodeNetworkId.value!][index][TX_DIRECTION.sent] = true;
+    }
+  }
+
+  function removePendingTransactionByAccount(address: Encoded.AccountAddress, hash: string) {
+    const pendingTransactionForAccount: ITransaction[] = (
+      transactions.value[address]?.pending?.[nodeNetworkId.value!] || []
+    );
+
+    if (pendingTransactionForAccount.length) {
+      transactions.value[address].pending[nodeNetworkId.value!] = (
+        pendingTransactionForAccount.filter((transaction) => transaction.hash !== hash)
+      );
+    }
+  }
+
+  async function waitTransactionMined(address: Encoded.AccountAddress, hash?: string) {
+    if (hash) {
+      try {
+        const aeSdk = await getAeSdk();
+        await aeSdk.poll(hash as Encoded.TxHash);
+        setPendingTransactionSentByHash(address, hash);
+      } catch (e) {
```

Comment:
you are not being consistent with the error variable name - you should stick either to `e` or `error` but I think in most places we use the second one